libs/output-mapping/src/Storage/TableStructureValidator.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Storage;
+
+use Keboola\OutputMapping\Exception\InvalidOutputException;
+use Keboola\OutputMapping\Exception\InvalidTableStructureException;
+use Keboola\OutputMapping\Mapping\MappingFromConfigurationSchemaColumn;
+use Keboola\OutputMapping\Writer\Helper\PrimaryKeyHelper;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\Utils\Sanitizer\ColumnNameSanitizer;
+use Psr\Log\LoggerInterface;
+
+class TableStructureValidator
+{
+    public function __construct(
+        readonly private bool $hasNewNativeTypeFeature,
+        readonly private LoggerInterface $logger,
+        readonly private Client $client,
+    ) {
+    }
+
+    /**
+     * @param MappingFromConfigurationSchemaColumn[] $schemaColumns
+     */
+    public function validateTable(string $tableId, array $schemaColumns): void
+    {
+        if (!$this->hasNewNativeTypeFeature) {
+            return;
+        }
+
+        try {
+            $table = $this->client->getTable($tableId);
+        } catch (ClientException $e) {
+            if ($e->getCode() === 404) {
+                return;
+            }
+            throw new InvalidOutputException($e->getMessage(), $e->getCode(), $e);
+        }
+
+        if ($table['isTyped']) {
```

Comment:
🙏 testík když je tabulka ve storage netypovaná, tak to nesletí na validaci

Reply:
pokud jsem to správně včera pochopil, tak validace pro netypované tabulky se bude přidávat v dalším tasku

Reply:
jj pro netypovaný bude hned následovat a jelikož to mezitím nebudeme releasovat tak to nemá cenu přidávat a v dalším PR mazat

Reply:
ok